### PR TITLE
added ForgetPassword class to authentication/views.py

### PR DESCRIPTION
**Pull request by makarman**
added GET POST and PUT methods
added email sender in POST methods

## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVPYTP-12)

## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/authentication/views.py [hunk 1]
```diff
@@ -12,3 +19,51 @@ class UserView(View):
...
+                if email_validator(email):
+                    user = CustomUser.get_by_email(email=email)
+                    if user:
+                        arg = {'user_id': user.id}
+                        token = create_token(arg)
+                        ctx = {'first_name': user.first_name,
+                               'token': token,
+                               'domain': FRONT_HOST}
+                        subject = 'Paswort reset'
+                        message = 'You tried change your password.'
```
Comment (Byte4Craft) [inline eventually/authentication/views.py]: "tried TO change"

File: eventually/authentication/views.py [hunk 2]
```diff
@@ -12,3 +19,51 @@ class UserView(View):
...
+                if user:
+                    data = json_loads(data=request.body)
+                    if required_keys_validator(data, ['new_password'], False):
+                        new_password = data.get('new_password')
+                        if password_validator(new_password):
+                            user.password = new_password
+                            user.save()
+                            ctx = {'first_name': user.first_name}
+                            subject = 'Paswort reset'
+                            message = 'Succesful paswort reset.'
```
Comment (Byte4Craft) [inline eventually/authentication/views.py]: "succesSful" and "passworD"

Comment (makarman) [reply eventually/authentication/views.py]: Done

File: eventually/authentication/views.py (hunk 1 above)
Comment (makarman) [reply eventually/authentication/views.py]: Done

**Comment by lhalam** [comment]
Resolve conflicts

**Review by ssOleg — APPROVED** [review]

**Pull request merged**